prusti-viper/src/encoder/mir_encoder.rs

```diff
@@ -17,11 +17,68 @@ use rustc_middle::{mir, ty};
 use rustc_index::vec::{Idx, IndexVec};
 use rustc_span::{Span, DUMMY_SP};
 use log::{trace, debug};
-use std::collections::HashMap;
+use std::{collections::HashMap, fmt::Display};
 
 pub static PRECONDITION_LABEL: &'static str = "pre";
 pub static WAND_LHS_LABEL: &'static str = "lhs";
 
+/// Result of encoding a place
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub enum PlaceEncoding {
+    /// Just an expression, the most common case.
+    Expr(vir::Expr),
+    /// Field access expression
+    FieldAccess { base: Box<PlaceEncoding>, field: vir::Field },
+    /// Array access expression
+    ArrayAccess { base: Box<PlaceEncoding>, index: Box<PlaceEncoding>, typ: vir::Type },
```

Comment:
Could the index be just `vir::Expr`? I remember we checked whether `x[y[0]]` is encoded as one or two MIR statement during the meeting, but I don't recall the outcome.

Reply:
yep, index is just a `vir::Expr`. as the [`ProjectionElem::Index(V)`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/enum.ProjectionElem.html) for `mir::Place` is instantiated with [`V` = `Local`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/type.PlaceElem.html) i think we can rely on that pretty safely.